File: src/org/cerion/symcalc/expression/GraphicsExpr.java
package org.cerion.symcalc.expression;

public class GraphicsExpr extends Expr {

    @Override
    public String toString() {
        return "Graphics";
    }

    @Override
    public void show(int i) {
        System.out.println("Graphics");
    }

    @Override
    public boolean equals(Expr e) {
        return false;
    }

    @Override
    protected Expr evaluate() {
        return this;
    }

    @Override
    protected ExprType getType() {
        return ExprType.GRAPHICS;
    }
}

File: src/org/cerion/symcalc/expression/GraphicsExpr.kt
package org.cerion.symcalc.expression

open class GraphicsExpr : Expr() {
    override val type: ExprType
        get() = ExprType.GRAPHICS

    override fun toString(): String = "Graphics"
    override fun show(i: Int) = println("Graphics")
    override fun equals(e: Expr): Boolean = false
    override fun evaluate(): Expr = this
}
